tests/test_backend_local.py

```diff
@@ -28,222 +28,194 @@
-
-
-def test_maildir_lists_two_messages_from_mailbox(tmpdir):
-	path = str(tmpdir.join('sample'))
-	sample_maildir = mailbox.Maildir(path, create=True)
-	add_maildir_message(sample_maildir, 'blaa-blaa-1', '', 'new')
-	add_maildir_message(sample_maildir, 'blaa-blaa-2', '', 'cur')
-	add_maildir_message(sample_maildir, 'blaa-blaa-3', 'S', 'cur')
-	sample_maildir.close()
-
-	be = create_backend('maildir', name='sample', path=path)
-	be.open()
-	try:
-		msgs = list(be.list_messages())
-		folders = [folder for folder, msg in msgs]
-		msg_ids = set(msg.get('message-id') for folder, msg in msgs)
-	finally:
+	def test_mailbox_should_be_closed(self, sample_path):
+		be = create_backend('maildir', path=sample_path)
+		be.open()
 		be.close()
-	assert len(msgs) == 2
-	assert all(folder == '' for folder in folders)
-	assert msg_ids == set(['blaa-blaa-1', 'blaa-blaa-2'])
-
+		assert not be.is_open()
 
-def test_maildir_should_have_folders(tmpdir):
-	path = str(tmpdir.join('sample'))
-	sample_maildir = mailbox.Maildir(path, create=True)
-	f = sample_maildir.add_folder('folder1')
-	sample_maildir.add_folder('folder2')
-	f.add_folder('subfolder')
-
-	be = create_backend('maildir', path=path)
-	be.open()
-	folders = be.request_folders()
-	assert set(['', 'folder1', 'folder2', 'folder1/subfolder']) == set(folders)
-
-
-def test_maildir_does_not_support_notifications(tmpdir): # for now
-	path = str(tmpdir.join('sample'))
-	sample_maildir = mailbox.Maildir(path, create=True)
-	be = create_backend('maildir', path=path)
-	be.open()
-	with pytest.raises(NotImplementedError):
-		be.notify_next_change()
-	with pytest.raises(NotImplementedError):
-		be.cancel_notifications()
-
-
-def test_maildir_open_should_fail_if_mailbox_does_not_exist(tmpdir):
-	path = str(tmpdir.join('not-exist'))
-
-	be = create_backend('maildir', path=path)
-	with pytest.raises(IOError):
+	def test_lists_no_messages_from_empty_mailbox(self, sample_path):
+		be = create_backend('maildir', name='sample', path=sample_path)
 		be.open()
+		try:
+			msgs = list(be.list_messages())
+			assert len(msgs) == 0
+		finally:
+			be.close()
+
+	def test_lists_unread_messages_from_mailbox(self, sample_path, sample_maildir):
+		self._add_message(sample_maildir, 'blaa-blaa-1', None, 'new')
+		self._add_message(sample_maildir, 'blaa-blaa-2', None, 'cur')
+		self._add_message(sample_maildir, 'blaa-blaa-3', 'S', 'cur')
+		sample_maildir.close()
+
+		be = create_backend('maildir', name='sample', path=sample_path)
+		be.open()
+		try:
+			msgs = list(be.list_messages())
+			folders = [folder for folder, msg in msgs]
+			msg_ids = set(msg.get('message-id') for folder, msg in msgs)
+		finally:
+			be.close()
+		assert len(msgs) == 2
+		assert all(folder == '' for folder in folders)
+		assert msg_ids == set(['blaa-blaa-1', 'blaa-blaa-2'])
+
+	def test_folders_should_be_listed(self, sample_path, sample_maildir):
+		f = sample_maildir.add_folder('folder1')
+		sample_maildir.add_folder('folder2')
+		f.add_folder('subfolder')
```

Comment:
It should, since it belongs to mailbox interface. However, maildir close does nothing, because maildir keeps no files open or locks between operations. I will fix.